core-bundle/src/Resources/contao/dca/tl_page.php

```diff
@@ -1122,153 +1128,46 @@ public function scheduleUpdate(DataContainer $dc)
 	 */
 	public function generateAlias($varValue, DataContainer $dc)
 	{
-		$objPage = PageModel::findWithDetails($dc->id);
-
-		$aliasExists = function (string $alias) use ($dc, $objPage): bool
-		{
-			$objAliasIds = $this->Database->prepare("SELECT id FROM tl_page WHERE alias=? AND id!=?")
-										  ->execute($alias, $dc->id);
-
-			if (!$objAliasIds->numRows)
-			{
-				return false;
-			}
-
-			$strCurrentDomain = $objPage->domain;
-			$strCurrentLanguage = $objPage->rootLanguage;
-
-			if ($objPage->type == 'root')
-			{
-				$strCurrentDomain = Input::post('dns');
-				$strCurrentLanguage = Input::post('language');
-			}
-
-			while ($objAliasIds->next())
-			{
-				$objAliasPage = PageModel::findWithDetails($objAliasIds->id);
-
-				if ($objAliasPage->domain != $strCurrentDomain)
-				{
-					continue;
-				}
-
-				if ($objAliasPage->rootLanguage != $strCurrentLanguage && Config::get('addLanguageToUrl'))
-				{
-					continue;
-				}
-
-				// Duplicate alias found
-				return true;
-			}
-
-			return false;
-		};
-
-		// Generate an alias if there is none
-		if ($varValue == '')
-		{
-			$varValue = System::getContainer()->get('contao.slug')->generate
-			(
-				$dc->activeRecord->title,
-				$dc->activeRecord->id,
-				static function ($alias) use ($objPage, $aliasExists)
-				{
-					return $aliasExists(($objPage->useFolderUrl ? $objPage->folderUrl : '') . $alias);
-				}
-			);
-
-			// Generate folder URL aliases (see #4933)
-			if ($objPage->useFolderUrl && $objPage->folderUrl != '')
-			{
-				$varValue = $objPage->folderUrl . $varValue;
-			}
-		}
-		elseif ($aliasExists($varValue))
-		{
-			throw new Exception(sprintf($GLOBALS['TL_LANG']['ERR']['aliasExists'], $varValue));
-		}
-
-		if ($varValue != $dc->activeRecord->alias)
-		{
-			$this->purgeSearchIndex($dc);
-		}
+		@trigger_error('Using "tl_page::generateAlias()" has been deprecated and will no longer work in Contao 5.0. Use "PageUrlListener::generateAlias()" instead.', E_USER_DEPRECATED);
```

Comment:
Same here: Do not use this code anyway.

Reply:
Changed in 1f70bdbeacefb095fa911bd2930bbcdd9ed709fd